src/gui/rss/rsswidget.cpp

```diff
@@ -54,6 +54,50 @@
 #include "feedlistwidget.h"
 #include "ui_rsswidget.h"
 
+namespace
+{
+    void convertRelativeUrlToAbsolute(QString &html, const QString &baseUrl);
+    
+    void convertRelativeUrlToAbsolute(QString &html, const QString &baseUrl)
+    {
+        const QRegularExpression rx {uR"(((<a\s+[^>]*?href|<img\s+[^>]*?src)\s*=\s*["'])((https?|ftp):)?(\/\/[^\/]*)?(\/?[^\/"].*?)(["']))"_s
+            , QRegularExpression::CaseInsensitiveOption};
+
+        const QString normalizedBaseUrl = baseUrl.endsWith(u'/') ? baseUrl : baseUrl + u'/';
+        const QUrl url {normalizedBaseUrl};
+        const QString defaultScheme = url.scheme();
+        QRegularExpressionMatchIterator iter = rx.globalMatch(html);
+
+        while (iter.hasNext())
+        {
+            const QRegularExpressionMatch match = iter.next();
+            const QString scheme = match.captured(4);
+            const QString host = match.captured(5);
+            if (!scheme.isEmpty())
+            {
+                if (host.isEmpty())
+                    break; // invalid URL, should never happen
+
+                // already absolute URL
+                continue;
+            }
+
+            QString relativePath = match.captured(6);
+            if (relativePath.startsWith(u'/'))
+                relativePath = relativePath.mid(1);
+
+            const QString absoluteUrl = !host.isEmpty()
+                    ? QString(defaultScheme + u":" + host) : QString(normalizedBaseUrl + relativePath);
```

Comment:
```suggestion
                    ? (defaultScheme + u':' + host) : (normalizedBaseUrl + relativePath);
```
or below if the above doesn't work for you:
```suggestion
                    ? QString(defaultScheme + u':' + host) : (normalizedBaseUrl + relativePath);
```

Reply:
The second form works in my compiler

Reply:
> (defaultScheme + u':' + host)

Just for fun, what is error message about this?

Reply:
`Incompatible operand types ('QStringBuilder<QStringBuilder<const QString &, char16_t>, const QString &>' and 'QStringBuilder<const QString &, QString &>')` from vscode clang plugin, and the compiler error is similar.

Reply:
```
error: incompatible operand types ('QStringBuilder<QStringBuilder<const QString &, char16_t>, const QString &>' and 'QStringBuilder<const QString &, QString &>')
[build]    88 |                     ? ((defaultScheme + u':') + host) : (normalizedBaseUrl + relativePath);
[build]       |                     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Also generates error when given a brace, seems to be missing templates in QT libraries.